### PR TITLE
add example for custom type conversion. Fix bug in per-invocation type conversion. Add fluent calls to ChatHistory

### PR DESCRIPTION
**Pull request by johnoliver**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,15 @@
+# 1.2.1
+
+- Fix bug in `FunctionInvocation` not using per-invocation type conversion when calling `withResultType`.
```
Comment (dsgrieve) [inline CHANGELOG.md:3]: Maybe we should start using jreleaser to keep the changelog up to date.

File: semantickernel-api/src/main/java/com/microsoft/semantickernel/contextvariables/converters/ContextVariableJacksonConverter.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.semantickernel.contextvariables.ContextVariableTypeConverter;
+import com.microsoft.semantickernel.contextvariables.ContextVariableTypeConverter.Builder;
+import com.microsoft.semantickernel.exceptions.SKException;
+
+/**
+ * A utility class for creating {@link ContextVariableTypeConverter} instances that use Jackson for
+ * serialization and deserialization.
+ */
+public final class ContextVariableJacksonConverter {
```
Comment (dsgrieve) [inline semantickernel-api/src/main/java/com/microsoft/semantickernel/contextvariables/converters/ContextVariableJacksonConverter.java:14]: `ContextVariableJacksonConverter` or `ContextVariableJsonConverter`? I mean, isn't this a JSON converter that just so happens to use Jackson?

**Review by dsgrieve — APPROVED** [review]
Should have done the fluent chat completion in a separate PR.

**Review by milderhc — APPROVED** [review]

File: semantickernel-api/src/main/java/com/microsoft/semantickernel/contextvariables/converters/ContextVariableJacksonConverter.java (hunk 2 above)
Comment (johnoliver) [reply semantickernel-api/src/main/java/com/microsoft/semantickernel/contextvariables/converters/ContextVariableJacksonConverter.java:14]: It does make a bit of a difference as typically you will need your classes to have Jackson annotations for this to work

**Pull request merged**